src/ark/segmentation/fiber_segmentation.py

```diff
@@ -188,8 +191,64 @@ def run_fiber_segmentation(data_dir, fiber_channel, out_dir, img_sub_folder=None
     return fiber_object_table
 
 
+def calculate_fiber_alignment(fiber_object_table, k=4, axis_thresh=2):
+    """ Calculates an alignment score for each fiber in an image. Based on the angle difference of
+    the fiber compared to it's k nearest neighbors.
+
+    Args:
+        fiber_object_table (pd.DataFrame):
+            dataframe containing the fiber objects and their properties (fov, label, alignment,
+             centroid-0, centroid-1, major_axis_length, minor_axis_length)
+        k (int):
+            number of neighbors to check alignment difference for
+        axis_thresh (int):
+            threshold for how much longer the length of the fiber must be compared to the width
+
+    Returns:
+        pd.DataFrame:
+         - Dataframe with the alignment scores appended
+    """
+
+    fovs = np.unique(fiber_object_table.fov)
+    fov_data = []
+
+    # process one fov at a time
+    for fov in fovs:
+        fov_fiber_table = fiber_object_table[fiber_object_table.fov == fov]
+
+        # only grab fibers of specified length to width ratio
+        filtered_lengths = fov_fiber_table[(fov_fiber_table['major_axis_length'].values /
+                                            fov_fiber_table['minor_axis_length'].values)
+                                           >= axis_thresh]
+        filtered_lengths = filtered_lengths.reset_index()
+
+        # create a distance matrix between fiber centroids
+        centroids = np.vstack((filtered_lengths['centroid-0'].values,
+                               filtered_lengths['centroid-1'].values)).T
+        fiber_dist_mat = cdist(centroids, centroids)
```

Comment:
Just for clarification, this is Euclidian distance between the list of 2d centroids correct?

Reply:
yep